src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -166,8 +169,48 @@ ciEnv::ciEnv(CompileTask* task)
   _jvmti_can_access_local_variables = false;
   _jvmti_can_post_on_exceptions = false;
   _jvmti_can_pop_frame = false;
+
+  _dyno_klasses = NULL;
+  _dyno_locs = NULL;
+  _dyno_name[0] = '\0';
 }
 
+class LocPusher {
```

Comment:
This needs big comment explaining what this does and how it helps replay.
And I don't like the name. May be `RecordLocaiton`.